SimCalorimetry/PhaseIIAnalyzer/plugins/PhaseIIAnalyzer.cc

```diff
@@ -0,0 +1,277 @@
+    EBEnergyHisto[isample] =
+        fs->make<TH1I>(Form("EnergyEB_%d", isample), Form("Energy sample %d  Barrel;ADC", isample), 950, 100, 2000);
+    EBGainHisto[isample] =
+        fs->make<TH1I>(Form("GainEB_%d", isample), Form("Gain Barrel sample %d;Gain", isample), 5, 0, 4);
+  }
+
+  SingleChannelE = fs->make<TH1I>("SingleChannelE", "Energy single channel Barrel;ADC", 17, -0.5, 16.5);
+  SingleChannelELow = fs->make<TH1I>("SingleChannelELow", "Energy single channel Barrel;ADC", 17, -0.5, 16.5);
+
+  Char_t histo[200];
+
+  sprintf(histo, "EcalDigiTaskBarrelOccupancy");
+  meEBDigiOccupancy_ = fs->make<TH2D>(histo, histo, 360, 0., 360., 170, -85., 85.);
+
+  sprintf(histo, "EcalDigiTaskBarrelOccupancyHigh");
+  meEBDigiOccupancyHigh_ = fs->make<TH2D>(histo, histo, 360, 0., 360., 170, -85., 85.);
+
+  sprintf(histo, "EcalDigiTaskBarrelOccupancyLow");
+  meEBDigiOccupancyLow_ = fs->make<TH2D>(histo, histo, 360, 0., 360., 170, -85., 85.);
+
+  sprintf(histo, "EcalDigiTaskBarrelDigisMultiplicity");
+  meEBDigiMultiplicity_ = fs->make<TH1D>(histo, histo, 612, 0., 61200);
+
+  for (int i = 0; i < ecalPh2::sampleSize; i++) {
+    sprintf(histo, "EcalDigiTaskBarrelAnalogPulse%02d", i + 1);
+    meEBDigiADCAnalog_[i] = fs->make<TH1D>(histo, histo, 4000, 0., 400.);
+
+    sprintf(histo, "EcalDigiTaskBarrelADCPulse%02dGain0_Saturated", i + 1);
+    meEBDigiADCgS_[i] = fs->make<TH1D>(histo, histo, 4096, -0.5, 4095.5);
+    sprintf(histo, "EcalDigiTaskBarrelGainPulse%02d", i + 1);
+    meEBDigiGain_[i] = fs->make<TH1D>(histo, histo, 4, 0, 4);
+  }
+
+  sprintf(histo, "EcalDigiTaskBarrelPedestalForPreSample");
+  meEBPedestal_ = fs->make<TH1D>(histo, histo, 4096, -0.5, 4095.5);
+
+  sprintf(histo, "EcalDigiTaskBarrelMaximumPositionGt100ADC");
+  meEBMaximumgt100ADC_ = fs->make<TH1D>(histo, histo, 10, 0., 10.);
+
+  sprintf(histo, "EcalDigiTaskBarrelMaximumPositionGt10ADC");
+  meEBMaximumgt10ADC_ = fs->make<TH1D>(histo, histo, 10, 0., 10.);
+
+  //   sprintf (histo, "EcalDigiTask Barrel ADC counts after gain switch" ) ;
+  //meEBnADCafterSwitch_ = fs->make<TH1D>histo, histo, 10, 0., 10.) ;
+}
+
+PhaseIIAnalyzer::~PhaseIIAnalyzer() {
+  // do anything here that needs to be done at desctruction time
+  // (e.g. close files, deallocate resources etc.)
+}
+
+//
+// member functions
+//
+
+// ------------ method called for each event  ------------
+void PhaseIIAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  //LogInfo("PhaseII") << "new event ";
+
+  Handle<EBDigiCollectionPh2> pDigiEB;
+  iEvent.getByToken(digiTokenEB_, pDigiEB);
+
+  // edm::ESHandle<EcalIntercalibConstantsMC> ical;
+  // iSetup.get<EcalIntercalibConstantsMCRcd>().get(ical);
+  // const EcalIntercalibConstantsMC* ical_map = ical.product();
+  // EcalIntercalibConstantsMC::const_iterator itical = ical_map->getMap().find(2);
+  // cout << "intercalib: " << (*itical) << endl;
+
+  const int MAXSAMPLES = ecalPh2::sampleSize;
```

Comment:
This variable seems unused. Further down `EBDataFrame::MAXSAMPLES` is used instead.